src/commands/CommandUtils.ts

```diff
@@ -34,11 +34,10 @@ export class CommandUtils {
         }
 
         const dataSourceExports = []
-        for (let fileExport in dataSourceFileExports) {
-            if (
-                InstanceChecker.isDataSource(dataSourceFileExports[fileExport])
-            ) {
-                dataSourceExports.push(dataSourceFileExports[fileExport])
+        for (const fileExport of dataSourceFileExports) {
+            const awaitedFileExport = await fileExport
```

Comment:
can't we add a check (something like `if (fileExport instanceof Promise)`) in order to make it more clear? also I recommend to add a comment line and explain why this change is needed.

Reply:
@pleerock Sure, we can. It was decided to do so for sake of shortness (as far as `await` does not make (almost) any effect on non-`Promise` value this way).

But if it will be more clear this way, we can do like this (and the comment would be like that):
```typescript
// It is necessary to await here in case of the exported async value (Promise<DataSource>).
// e.g. the DataSource is instantiated with an async factory in the source file
const awaitedFileExport = fileExport instanceof Promise 
  ? await fileExport
  : fileExport;
```

What do you think?

Reply:
@pleerock It's possible, but some people still use eg. Bluebird promises or compilation chains with `Promise` polyfills. There isn't any benefit in `instanceof Promise` checks, but only some obscure bug reports.

Reply:
I think `await`ing doesn't introduce any overhead, as MDN states - Returns the fulfilled value of the promise, or the value itself if it's not a Promise. And even if there is any, it is too small. So I would agree with @Ginden on this one